imgui_widgets.cpp

```diff
@@ -4630,12 +4694,22 @@ void ImGui::ColorTooltip(const char* text, const float* col, ImGuiColorEditFlags
     }
 
     ImVec2 sz(g.FontSize * 3 + g.Style.FramePadding.y * 2, g.FontSize * 3 + g.Style.FramePadding.y * 2);
-    ColorButton("##preview", ImVec4(col[0], col[1], col[2], col[3]), (flags & (ImGuiColorEditFlags_NoAlpha | ImGuiColorEditFlags_AlphaPreview | ImGuiColorEditFlags_AlphaPreviewHalf)) | ImGuiColorEditFlags_NoTooltip, sz);
+    ColorButton("##preview", ImVec4(col[0], col[1], col[2], flags & ImGuiColorEditFlags_NoAlpha ? 1.0f : col[3]), (flags & (ImGuiColorEditFlags__InputMask | ImGuiColorEditFlags_NoAlpha | ImGuiColorEditFlags_AlphaPreview | ImGuiColorEditFlags_AlphaPreviewHalf)) | ImGuiColorEditFlags_NoTooltip, sz);
```

Comment:
This is the only change I don't understand, why did you change the `col[3]` with `flags & ImGuiColorEditFlags_NoAlpha ? 1.0f : col[3]` here?

(please don't submit a force-pushed PR because I already have a locally modified PR)

Reply:
I intended to break this out into a different bugfix PR and forgot, sorry. It's unrelated to the change in this PR but it's a bug I noticed as I was working on this stuff. According to [the comment on this function](https://github.com/ocornut/imgui/pull/2384/files/c907e1dfe1210b0b5a106ce3a03f8220ed865116#diff-da12ab9aaac3f0f311ef47c908abaaa0R4681) and [the docs on the `NoAlpha` flag](https://github.com/ocornut/imgui/blob/f717df4eb6892cde25c711e76abcdc72db4782e3/imgui.h#L1109), if `NoAlpha` is set, we aren't supposed to access `col[3]`; this modifies the behavior to not perform the access the docs say this function shouldn't do.

Reply:
Oooh, you are totally right! Definitively a bug fix then! Thanks!